controllers/datadogagent/feature/dummy/feature.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package dummy
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+)
+
+func init() {
+	err := feature.Register(feature.DummyIDType, buildDummyfeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func buildDummyfeature(options *feature.Options) feature.Feature {
+	return &dummyFeature{}
+}
+
+const (
+	dummyLabelKey   = "feature.com/dummy"
+	dummyLabelValue = "true"
+)
+
+type dummyFeature struct{}
+
+func (f *dummyFeature) Configure(dda *v2alpha1.DatadogAgent) bool {
+	return false
+}
+
+func (f *dummyFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) bool {
+	return false
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *dummyFeature) ManageDependencies(store feature.DependenciesStoreClient) error {
+	return nil
```

Comment:
With the store API, it means each deps need to do:
```
Get -> Check if nil -> Not nil: Update
                    -> Nil: Create
```

I think we should think about how to improve this experience before implementing features?

Reply:
IMO dependencies are not so often shared between features.

I was thinking to create a function `GetOrCreate` that return the instance or create a new instance automatically.
I'm open to other idea

Reply:
I was thinking about `Roles` for instance (or `ConfigMap` if we decide to move all configs to `datadog.yaml`). Having a `GetOrCreate` may not be useful as you will still need to check if the object is newly created or not to fill some fieds.

I think we way want to have `Managers` on top of the store. Like the `RBACManager`, offering a high level API like `AddPermissions(serviceAccount,  roleName, permissions)` and when `Apply` is called on the store, it lazily generates the necessary objects.

About the comment below, your intention looks like a `PodTemplateManager` to me, with its helper functions to manage each piece.

Reply:
@vboulineau 
I have added a new commit with a `RBACManager` https://github.com/DataDog/datadog-operator/pull/412/commits/b0722e26dd2c1ca1aaccaf7b663a83c589e9a8d6